[Fix-6220] [API] New tenant name could be "default"

LiuBodong opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler. Please review https://dolphinscheduler.apache.org/en-us/community/development/pull-request.html before opening a pull request.-->


## Purpose of the pull request
Fix bug #6220 

## Brief change log
When create a new Tenant, check if it's tenantCode equals "default"

resolves #6220

Commit:
[StandaloneServer][BUG]Fix workflow can't save correctly (#6214)

Commit:
Merge remote-tracking branch 'origin/dev' into dev

Commit:
[StandaloneServer][BUG] Command can't be saved to h2 database (#6214)

Commit:
[API][BUG] Fix New Tenant name could be "default" (#6220)

ruanwenjun commented on sql/dolphinscheduler_h2.sql:
```diff
@@ -315,6 +315,7 @@ CREATE TABLE t_ds_command
     id                        int(11) NOT NULL AUTO_INCREMENT,
     command_type              tinyint(4) DEFAULT NULL,
     process_definition_id     int(11) DEFAULT NULL,
```
Can we remove this field, it seems `process_definition_id` has not been used?

LiuBodong replied:
Yes, I will do it

LiuBodong committed:
Merge branch 'apache:dev' into bugfix-6220

LiuBodong committed:
Revert changes

Revert unnecessary changes not belongs to this issue

brave-lee commented on dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/impl/TenantServiceImpl.java:
```diff
@@ -95,7 +95,8 @@
 
-        if (checkTenantExists(tenantCode)) {
+        // check if tenant exists in database or named "default"
+        if (checkTenantExists(tenantCode) || "default".equalsIgnoreCase(tenantCode)) {
```
Why not allow tenants to be set to 'default' ?

LiuBodong replied:
![图片](https://user-images.githubusercontent.com/23203149/133557913-946ed17e-e519-4a5f-8dae-9e9f0525ddc9.png)

There's already a tenant named "default" which is a default tenant if no other tenant could be chosen, if we create a new tenant named "default", it hard to figure out which one is the "real" tenant we need.

brave-lee replied:
I think the front-end display should be repaired here. When there is already a tenant, the virtual tenant should not be displayed

LiuBodong replied:
Yes, you are right.
I think the front-end should display noting if no tenant was created.
if the user want to create a workflow when no tenant exists, he should create a new tenant first.

brave-lee replied:
I have different suggestion. if the user want to create a workflow when no tenant exists, in order not to affect workflow creation the tenant id will be -1, and now the tenant code is 'default'. I think that's how default appears.

LiuBodong replied:
So the question is how to avoid 'default' tenant in front-end and user created tenant named 'default' both appears.
Should we add a mark in the front-end to distinguish? or just remove the 'default' option when tenant list is not empty.

brave-lee replied:
I mean, if the tenant exists, it will be displayed, and the default of this enumeration will not be displayed. If it does not exist, the default of enumeration will be displayed

Commit:
Merge branch 'dev' of https://github.com/apache/dolphinscheduler into bugfix-6220

Commit:
[API][BUG] Fix new Tenant name could be "default" (#6220)

(1) filter 'default' tenant code in front-end instead of back-end
(2) front-end: check if the tenant list contains more than one tenant, if true, maybe the tenant list contains 'default' tenant, so we need to filter it out.
